packages/spearly-cms-js-core/src/Generator.ts

```diff
@@ -138,57 +138,74 @@ export class SpearlyJSGenerator {
-                if (!contentType) {
-                    // Error case.
-                    return Promise.reject(new Error("cms-loop element doesn't have cms-field"))
-                }
-                const varName = node.getAttribute("cms-item-variable")
-                node.removeAttribute("cms-loop")
-                node.removeAttribute("cms-field")
-                node.removeAttribute("cms-item-variable")
-                const generatedStr = await this.generateList(node.outerHTML, contentType, varName || contentType, apiOptions, insertDebugInfo)
-                const generatedNode = parse(generatedStr) as HTMLElement
-                resultNode.childNodes = generatedNode.childNodes
-                continue
-            }
-            if (node.childNodes.length > 0) {
-                node.childNodes = await this.traverseInjectionSubLoop(node.childNodes as HTMLElement[], apiOptions, insertDebugInfo)
-            }
-            resultNode.appendChild(node)
+    getElementDepth(targetElement: HTMLElement, topElement: HTMLElement) {
+        let depth = 0
+        let currentElement = targetElement
+        while (currentElement !== topElement) {
+            depth++
+            currentElement = currentElement.parentNode as HTMLElement
         }
-        return resultNode.childNodes
+        return depth
     }
 
-    async generateSubLoop(templateHtml: string, apiOptions: APIOption, insertDebugInfo): Promise<string> {
-        const parsedNode = parse(templateHtml)
-        parsedNode.childNodes = await this.traverseInjectionSubLoop(parsedNode.childNodes as HTMLElement[], apiOptions, insertDebugInfo)
-        return parsedNode.outerHTML
+    async generateSubLoop(nodeHTML: string, contentTypeId: string, insertDebugInfo: boolean, parentData: Content): Promise<string> {
+        const node = parse(nodeHTML);
+        const cmsLoopNodes = node.querySelectorAll("[cms-loop]");
+
+        if (cmsLoopNodes.length > 0) {
+            // sort cms loop nodes by depth from top element with getElementDepth.
+            const sortedCmsLoopNodes = cmsLoopNodes.sort((a, b) => {
+                return this.getElementDepth(b, node) - this.getElementDepth(a, node)
+            })
+            for (const cmsLoopNode of sortedCmsLoopNodes) {
+                const referenceFieldId = cmsLoopNode.getAttribute("cms-field")
+                const variableName = cmsLoopNode.getAttribute("cms-variable")
+                // Remove attributes
+                cmsLoopNode.removeAttribute("cms-loop")
+                cmsLoopNode.removeAttribute("cms-field")
+                cmsLoopNode.removeAttribute("cms-variable")
+
+                const contentTypeName = variableName || `${contentTypeId}_${referenceFieldId}`
+                const targetData = parentData.attributes.fields.data.filter(field => field.attributes.identifier === referenceFieldId);
+                if (targetData.length === 0) continue;
+
+                // @ts-ignore
```

Comment:
レビュー用コメント： `@spearly/sdk-js` の参照フィールドの定義で、`value.data` の定義がないためエラーとなっています。( https://github.com/unimal-jp/spearly-sdk-js/pull/87 で修正 PR を出しています)  
Comments for reviewing: This line will be lint error, because `@spearly/sdk-js` doesn't have definition of `value.data`.(I requested the fix PR for https://github.com/unimal-jp/spearly-sdk-js/pull/87)